x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/TransportUpdateConnectorFilteringAction.java

```diff
@@ -47,6 +52,23 @@ protected void doExecute(
         UpdateConnectorFilteringAction.Request request,
         ActionListener<ConnectorUpdateActionResponse> listener
     ) {
-        connectorIndexService.updateConnectorFiltering(request, listener.map(r -> new ConnectorUpdateActionResponse(r.getResult())));
+        String connectorId = request.getConnectorId();
+        List<ConnectorFiltering> filtering = request.getFiltering();
+        FilteringAdvancedSnippet advancedSnippet = request.getAdvancedSnippet();
+        List<FilteringRule> rules = request.getRules();
+        if (request.getFiltering() == null) {
+            connectorIndexService.updateConnectorFilteringDraft(
+                connectorId,
+                advancedSnippet,
+                rules,
```

Comment:
Could it happen that this list is empty and we remove the `default` rule on accident?

Reply:
Yup, added check to make sure that we check that we reference a `default` value